.cicd/scripts/srw_ftest.sh

```diff
@@ -0,0 +1,132 @@
+# NOTE: At this time, this script is a placeholder for functional test framework.
+# At this time, we are leaving the exercise of graphical plotting for a later stage, perhaps WE2E state.
+#
+[[ -n ${ACCOUNT} ]] || ACCOUNT="no_account"
+[[ -n ${BRANCH} ]] || BRANCH="develop"
+[[ -n ${TASKS} ]] || TASKS=""
+[[ -n ${TASK_DEPTH} ]] || TASK_DEPTH=4
+[[ -n ${FORGIVE_PYTHON} ]] || FORGIVE_PYTHON=false
+set -e -u -x
+
+script_dir="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" > /dev/null 2>&1 && pwd)"
+
+# Get repository root from Jenkins WORKSPACE variable if set, otherwise, set
+# relative to script directory.
+declare workspace
+if [[ -n "${WORKSPACE}" ]]; then
+    workspace="${WORKSPACE}"
+else
+    workspace="$(cd -- "${script_dir}/../.." && pwd)"
+fi
+
+# Normalize Parallel Works cluster platform value.
+declare platform
+if [[ "${SRW_PLATFORM}" =~ ^(az|g|p)clusternoaa ]]; then
+    platform='noaacloud'
+else
+    platform="${SRW_PLATFORM}"
+fi
+
+# Test directories
+we2e_experiment_base_dir="${workspace}/expt_dirs"
+we2e_test_dir="${workspace}/tests/WE2E"
+nco_dir="${workspace}/nco_dirs"
+
+pwd
+
+# The following essentially sets up and performs tasks described in the user documentation section:
+#     "Run the Workflow Using Stand-Alone Scripts".
+echo "BRANCH=${BRANCH}"
+
+# Set the ACCOUNT to use for this PLATFORM ...
+sed "s|^  ACCOUNT: \"\"|  ACCOUNT: \"${ACCOUNT}\"|1" -i ush/config_defaults.yaml
+sed "s|hera|${platform,,}|1" ush/config.community.yaml | sed "s|an_account|${ACCOUNT}|1" > ush/config.yaml
+
+# Set directory paths ...
+export EXPTDIR=${workspace}/expt_dirs/test_community
+echo "EXPTDIR=${EXPTDIR}"
+sed "s|^workflow:|workflow:\n  EXPT_BASEDIR: ${workspace}/expt_dirs|1" -i ush/config.yaml
+sed "s|^workflow:|workflow:\n  EXEC_SUBDIR: ${workspace}/install_${SRW_COMPILER}/exec|1" -i ush/config.yaml
+
+# DATA_LOCATION differs on each platform ... find it.
+export DATA_LOCATION=$(grep TEST_EXTRN_MDL_SOURCE_BASEDIR ${workspace}/ush/machine/${SRW_PLATFORM,,}.yaml | awk '{printf "%s", $2}')
+echo "DATA_LOCATION=${DATA_LOCATION}"
+
+# Configure a default test ...
+sed "s|^task_get_extrn_ics:|task_get_extrn_ics:\n  EXTRN_MDL_SOURCE_BASEDIR_ICS: ${DATA_LOCATION}/FV3GFS/grib2/2019061518|1" -i ush/config.yaml
+sed "s|^task_get_extrn_lbcs:|task_get_extrn_lbcs:\n  EXTRN_MDL_SOURCE_BASEDIR_LBCS: ${DATA_LOCATION}/FV3GFS/grib2/2019061518|1" -i ush/config.yaml
+
+[[ ${FORGIVE_PYTHON} == true ]] && set +e +u    # Some platforms have incomplete python3 or conda support, but wouldn't necessarily block workflow tests
+# Consistency check ...
+cd ${workspace}/ush
+        ./config_utils.py -c $(pwd)/config.yaml -v $(pwd)/config_defaults.yaml
+cd ${workspace}
+
+# Activate the workflow environment ...
+source etc/lmod-setup.sh ${platform,,}
+module use modulefiles
+module load build_${platform,,}_${SRW_COMPILER}
+module load wflow_${platform,,}
+conda activate regional_workflow
```

Comment:
The config_utils.py script requires python3 to be available before being called.  I moved the activation of the workflow environment to before line 68 and all that you should need to load the environment is:
`source ${workspace}/ush/load_modules_wflow.sh ${platform,,}`

Reply:
Great hint. Probably the documentation needs to be updated to reflect this better approach.

Reply:
Replacing the workflow environment with:
`source ${workspace}/ush/load_modules_wflow.sh ${platform,,}`
was necessary to keep the script from erroring out for me due to attempting to activate the regional_workflow conda environment.

Further testing showed that using this script is why the `run_orog` task was failing (the build modulefile is purged when calling the `load_modules_wflow.sh` script).  I was able to create a new script, `ush/load_modules_ftest.sh`, that loads the build modulefile, the wflow modulefile, then activates regional_workflow.  With this script, I was able to successfully run the `run_orog` wrapper script, passing the test.

Reply:
If you are interested in seeing what I did with the new script, please see:
`/work/noaa/epic-ps/mlueken/ufs-srweather-app/ush/load_modules_ftest.sh`